### PR TITLE
Correct result for passing DOMException.prototype to URLSearchParams

### PR DESCRIPTION
**Pull request by annevk**
See discussion in https://bugs.webkit.org/show_bug.cgi?id=166973.

### CODE REVIEW

File: url/urlsearchparams-constructor.html [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 test(() => {
     params = new URLSearchParams(DOMException.prototype);
-    assert_equals(params.toString(), "Error=")
+    assert_equals(params.toString(), "INDEX_SIZE_ERR=1&&HIERARCHY_REQUEST_ERR=3&WRONG_DOCUMENT_ERR=4&INVALID_CHARACTER_ERR=5&NO_MODIFICATION_ALLOWED_ERR=7&NOT_FOUND_ERR=8&NOT_SUPPORTED_ERR=9&INUSE_ATTRIBUTE_ERR=10&INVALID_STATE_ERR=11&SYNTAX_ERR=12&INVALID_MODIFICATION_ERR=13&NAMESPACE_ERR=14&SECURITY_ERR=18&NETWORK_ERR=19&ABORT_ERR=20&URL_MISMATCH_ERR=21&QUOTA_EXCEEDED_ERR=22&TIMEOUT_ERR=23&INVALID_NODE_TYPE_ERR=24&DATA_CLONE_ERR=25")
```
Comment (domenic) [inline url/urlsearchparams-constructor.html]: && after INDEX_SIZE_ERR=1?

Comment (annevk) [inline url/urlsearchparams-constructor.html]: Well spotted. Sorry for that.

**Comment by domenic** [comment]
LGTM but we should get @cdumez's review too I think

File: url/urlsearchparams-constructor.html [hunk 2]
```diff
@@ -24,7 +24,7 @@
 
 test(() => {
     params = new URLSearchParams(DOMException.prototype);
-    assert_equals(params.toString(), "Error=")
+    assert_equals(params.toString(), "INDEX_SIZE_ERR=1&HIERARCHY_REQUEST_ERR=3&WRONG_DOCUMENT_ERR=4&INVALID_CHARACTER_ERR=5&NO_MODIFICATION_ALLOWED_ERR=7&NOT_FOUND_ERR=8&NOT_SUPPORTED_ERR=9&INUSE_ATTRIBUTE_ERR=10&INVALID_STATE_ERR=11&SYNTAX_ERR=12&INVALID_MODIFICATION_ERR=13&NAMESPACE_ERR=14&SECURITY_ERR=18&NETWORK_ERR=19&ABORT_ERR=20&URL_MISMATCH_ERR=21&QUOTA_EXCEEDED_ERR=22&TIMEOUT_ERR=23&INVALID_NODE_TYPE_ERR=24&DATA_CLONE_ERR=25")
```
Comment (cdumez) [inline url/urlsearchparams-constructor.html]: FYI, Webkit nightly is getting:
INDEX_SIZE_ERR=1&DOMSTRING_SIZE_ERR=2&HIERARCHY_REQUEST_ERR=3&WRONG_DOCUMENT_ERR=4&INVALID_CHARACTER_ERR=5&NO_DATA_ALLOWED_ERR=6&NO_MODIFICATION_ALLOWED_ERR=7&NOT_FOUND_ERR=8&NOT_SUPPORTED_ERR=9&INUSE_ATTRIBUTE_ERR=10&INVALID_STATE_ERR=11&SYNTAX_ERR=12&INVALID_MODIFICATION_ERR=13&NAMESPACE_ERR=14&INVALID_ACCESS_ERR=15&VALIDATION_ERR=16&TYPE_MISMATCH_ERR=17&SECURITY_ERR=18&NETWORK_ERR=19&ABORT_ERR=20&URL_MISMATCH_ERR=21&QUOTA_EXCEEDED_ERR=22&TIMEOUT_ERR=23&INVALID_NODE_TYPE_ERR=24&DATA_CLONE_ERR=25

Comment (cdumez) [inline url/urlsearchparams-constructor.html]: So it looks like we have the following extras:
DOMSTRING_SIZE_ERR=2
NO_DATA_ALLOWED_ERR=6
INVALID_ACCESS_ERR=15
VALIDATION_ERR=16
TYPE_MISMATCH_ERR=17

Comment (cdumez) [inline url/urlsearchparams-constructor.html]: The Web IDL spec [1] still seems to contain the deprecated constants that WebKit supports. Am I missing something?

[1] https://heycam.github.io/webidl/#idl-DOMException-error-names

Comment (annevk) [inline url/urlsearchparams-constructor.html]: I think the problem is with IDL. It only talks about one table, while in fact there are two. And for defining properties order is somewhat important, but with two tables order is lost.

I filed https://github.com/heycam/webidl/issues/284 and will push an update to this test shortly.

**Comment by cdumez** [comment]
LGTM.

**Pull request merged**